src/middlewared/middlewared/plugins/iscsi.py

```diff
@@ -377,3 +379,363 @@ class Config:
+                    )
+
+            if (os.path.exists(path) and not
+                    os.path.isfile(path)) or path[-1] == '/':
+                verrors.add(f'{schema_name}.path',
+                            'You need to specify a filepath not a directory')
+
+            for p in pools:
+                mp_path = f'/mnt/{p["name"]}'
+                if path == mp_path:
+                    verrors.add(
+                        f'{schema_name}',
+                        'You need to specify a file inside your volume/dataset'
+                    )
+                if path.startswith(f'{mp_path}/'):
+                    valid = True
+
+            if not valid:
+                verrors.add(
+                    f'{schema_name}.path',
+                    'Your path to the extent must reside inside a'
+                    ' volume/dataset mountpoint.')
+
+        return data
+
+    @private
+    async def clean_size(self, data, schema_name, verrors):
+        extent_type = data['type']
+        path = data['path']
+        size = data['filesize']
+        blocksize = data['blocksize']
+
+        if (
+            size == 0 and path and (not os.path.exists(path) or (
+                os.path.exists(path) and not
+                os.path.isfile(path)
+            ))
+        ):
+            verrors.add(
+                f'{schema_name}.path',
+                'The file must exist if the extent size is set to auto (0)')
+        elif extent_type == 'file' and not path:
+            verrors.add(f'{schema_name}.path', 'This field is required')
+
+        if size and size != 0 and blocksize:
+            if (float(size) / blocksize) % 1 != 0:
+                verrors.add(f'{schema_name}.filesize',
+                            'File size must be a multiple of block size')
+
+        return data
+
+    @private
+    async def extent_serial(self, serial):
+        # TODO Just ported, let's do something different later? - Brandon
+        if serial is None:
+            try:
+                nic = (await self.middleware.call('interfaces.query',
+                                                  [['name', 'rnin', 'vlan'],
+                                                   ['name', 'rnin', 'lagg'],
+                                                   ['name', 'rnin', 'epair'],
+                                                   ['name', 'rnin', 'vnet'],
+                                                   ['name', 'rnin', 'bridge']])
+                       )[0]
+                mac = nic['link_address'].replace(':', '')
+
+                ltg = await self.query()
+                if len(ltg) > 0:
+                    lid = ltg[0]['id']
+                else:
+                    lid = 0
+                return f'{mac.strip()}{lid:02}'
+            except Exception:
+                return '10000001'
+        else:
+            return serial
+
+    @accepts()
+    async def populate_disk_choice(self):
```

Comment:
That does not sound like a good name for a public method that will be used by the UI.

How about `disk_choices` ?